Lesson4.2/src/com/home/model/MilitaryUnit.java

```diff
@@ -0,0 +1,74 @@
+package com.home.model;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class MilitaryUnit {
+
+    private final int maxNumberOfRecruits;
+    private int currentNumberOfRecruits;
+    private final List<Person> recruitList;
+
+    public MilitaryUnit(int maxNumberOfRecruits) {
+        this.maxNumberOfRecruits = maxNumberOfRecruits;
+        recruitList = new LinkedList<>();
+    }
+
+    public void addNewRecruit(Person person) {
+        if (currentNumberOfRecruits < maxNumberOfRecruits) {
+            if (MilitaryOffice.checkForSuitability(person) && checkForSimilarRecruit(person)) {
```

Comment:
I'd rename checkForSuitability to isPersonSuitable and checkForSimilarRecruit to isSimilarReqruitExists(in that case when this method would return true then person shouldn't be added to the  MilitaryUnit) or isPersonUnique to increase what exactly method returns, but this is optional